Remove Unnecessary Mixed Content Form Logic

ruthvikkonda opened the pull request:
#### 1ef28e14dde9685720165b46a45b5c4b8bb51fde
<pre>
Remove Unnecessary Mixed Content Form Logic
<a href="https://bugs.webkit.org/show_bug.cgi?id=298572">https://bugs.webkit.org/show_bug.cgi?id=298572</a>
<a href="https://rdar.apple.com/160159601">rdar://160159601</a>

Reviewed by Matthew Finkel.

After deprecating didDisplayInsecureContent() and didRunInsecureContent(),
checkFormForMixedContent() now does nothing besides emitting a console log
that is not useful by itself. This method can be safely removed. The logic
that the W3C spec specifies for mixed content form submissions is to display a
UI prompt for the user that allows them to abort the submission. This logic is
already in place in Safari so we satisfy spec requirements.
Similar to Safari, the embedding application should be implementing the relevant
logic for insecure form submissions.

No new behavior is introduced so new tests are added. Existing related tests are appropriately removed
since they are no longer relevant.

* LayoutTests/http/tests/security/mixedContent/insecure-form-in-iframe-expected.txt: Removed.
* LayoutTests/http/tests/security/mixedContent/insecure-form-in-iframe.html: Removed.
* LayoutTests/http/tests/security/mixedContent/javascript-url-form-in-main-frame-expected.txt: Removed.
* LayoutTests/http/tests/security/mixedContent/javascript-url-form-in-main-frame.html: Removed.
* LayoutTests/http/tests/security/mixedContent/resources/frame-with-insecure-form.html: Removed.
* LayoutTests/http/tests/security/mixedContent/resources/frame-with-javascript-url-form.html: Removed.
* LayoutTests/http/tests/security/originHeader/origin-header-for-https-expected.txt:
* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/platform/wk2/http/tests/security/mixedContent/insecure-form-in-iframe-expected.txt: Removed.
* LayoutTests/platform/wk2/http/tests/security/mixedContent/javascript-url-form-in-main-frame-expected.txt: Removed.
* Source/WebCore/history/CachedFrame.cpp:
* Source/WebCore/html/HTMLFormElement.cpp:
(WebCore::HTMLFormElement::attributeChanged):
* Source/WebCore/loader/MixedContentChecker.cpp:
(WebCore::MixedContentChecker::checkFormForMixedContent): Deleted.
* Source/WebCore/loader/MixedContentChecker.h:

Canonical link: <a href="https://commits.webkit.org/299785@main">https://commits.webkit.org/299785@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4e4597b6388f3404d941db0d57915c2dafd437bc

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120148 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39842 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30493 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126497 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72221 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/5da89f6a-6035-4638-888c-1c0cc4f5f679) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/122024 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40538 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48419 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91245 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60550 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/da15d4c2-f6a6-45a5-b323-d1c4b5bcfb84) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/123100 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32383 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107723 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71800 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7957edc6-479f-4551-9cd5-fda3d7e670d2) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31415 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25829 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70123 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101863 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26016 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129401 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/47068 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/35707 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99865 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47434 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103911 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99708 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25318 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45161 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23182 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/43699 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46930 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52636 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46398 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49745 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/48082 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

sysrqb commented on LayoutTests/http/tests/security/mixedContent/insecure-form-in-iframe.html:
```diff
@@ -1,14 +0,0 @@
-<p>This test loads a secure iframe that has a form with an insecure action.  We should
-*not* get a mixed content callback because the main frame is HTTP and the form
-doesn't contaminate the child iframe's security origin with mixed content.</p>
-<iframe src="https://127.0.0.1:8443/security/mixedContent/resources/frame-with-insecure-form.html"></iframe>
```
Looks like you can delete these resources, too

sysrqb commented on LayoutTests/http/tests/security/mixedContent/javascript-url-form-in-main-frame.html:
```diff
@@ -1,25 +0,0 @@
-by active network attackers.</p>
-<script>
-onload = function() {
-    window.open("https://127.0.0.1:8443/security/mixedContent/resources/frame-with-javascript-url-form.html");
```
And this one

sysrqb commented on Source/WebCore/loader/MixedContentChecker.cpp:
```diff
@@ -161,18 +161,4 @@ bool MixedContentChecker::shouldBlockRequest(LocalFrame& frame, const URL& url,
-        return;
-
-    auto message = makeString("The page at "_s, frame.document()->url().stringCenterEllipsizedToLength(), " contains a form which targets an insecure URL "_s, url.stringCenterEllipsizedToLength(), ".\n"_s);
-    frame.protectedDocument()->addConsoleMessage(MessageSource::Security, MessageLevel::Warning, message);
```
We could keep this console log, it doesn't hurt, but I don't know if it helps anyone at this point.

sysrqb commented:
Looks like `http/tests/security/originHeader/origin-header-for-https.html` is a real failure, at least, too - probably just rebaseline.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a86565286693dc6119fd6ce121b665373a1ea914. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50475

ruthvikkonda committed:
Remove Unnecessary Mixed Content Form Logic
https://bugs.webkit.org/show_bug.cgi?id=298572
rdar://160159601

Reviewed by Matthew Finkel.

After deprecating didDisplayInsecureContent() and didRunInsecureContent(),
checkFormForMixedContent() now does nothing besides emitting a console log
that is not useful by itself. This method can be safely removed. The logic
that the W3C spec specifies for mixed content form submissions is to display a
UI prompt for the user that allows them to abort the submission. This logic is
already in place in Safari so we satisfy spec requirements.
Similar to Safari, the embedding application should be implementing the relevant
logic for insecure form submissions.

No new behavior is introduced so new tests are added. Existing related tests are appropriately removed
since they are no longer relevant.

* LayoutTests/http/tests/security/mixedContent/insecure-form-in-iframe-expected.txt: Removed.
* LayoutTests/http/tests/security/mixedContent/insecure-form-in-iframe.html: Removed.
* LayoutTests/http/tests/security/mixedContent/javascript-url-form-in-main-frame-expected.txt: Removed.
* LayoutTests/http/tests/security/mixedContent/javascript-url-form-in-main-frame.html: Removed.
* LayoutTests/http/tests/security/mixedContent/resources/frame-with-insecure-form.html: Removed.
* LayoutTests/http/tests/security/mixedContent/resources/frame-with-javascript-url-form.html: Removed.
* LayoutTests/http/tests/security/originHeader/origin-header-for-https-expected.txt:
* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/platform/wk2/http/tests/security/mixedContent/insecure-form-in-iframe-expected.txt: Removed.
* LayoutTests/platform/wk2/http/tests/security/mixedContent/javascript-url-form-in-main-frame-expected.txt: Removed.
* Source/WebCore/history/CachedFrame.cpp:
* Source/WebCore/html/HTMLFormElement.cpp:
(WebCore::HTMLFormElement::attributeChanged):
* Source/WebCore/loader/MixedContentChecker.cpp:
(WebCore::MixedContentChecker::checkFormForMixedContent): Deleted.
* Source/WebCore/loader/MixedContentChecker.h:

Canonical link: https://commits.webkit.org/299785@main
